### PR TITLE
Move plone.app.caching documentation to deployment/caching

### PR DESCRIPTION
**Pull request by ericof**
- [X] Move existing documentation from `plone.app.caching`
- [X] Convert documentation from RST to Markdown
- [X] Remove outdated references to kss
- [X] Add am introduction about HTTP Caching

### CODE REVIEW

File: docs/deployment/caching/control-panel.md
Comment (sneridagh) [inline docs/deployment/caching/control-panel.md]: typo

File: docs/deployment/caching/composite-views.md
Comment (sneridagh) [inline docs/deployment/caching/composite-views.md:11]: myst meta missing?

**Review by sneridagh — DISMISSED** [review]
Minor typos, the rest LGTM

File: docs/deployment/caching/composite-views.md
Comment (jensens) [inline docs/deployment/caching/composite-views.md]: rephrased a bit, the original is hard to understand and not 100% correct.

```suggestion
The term *composite view* refers to most of the page views that are visible on a Plone website, including views of content listings, content folders, many template views, and rarely also specific content items. 

These views can be difficult to manage because of the various changes that may affect the final composite view. As a result, composite views are typically challenging to remove reliably from caching proxies. 
To address this issue, the default caching profiles set headers that expire the cache immediately, which is known as *weak caching*.

However, the inline resources linked to from the composite view, such as CSS, JavaScript, and images, can typically be cached effectively in proxies. As a result, caching proxies can improve the overall speed of most composite views, even if the page itself is not cached.

For relatively stable composite views or views where some staleness is tolerable, it may be tempting to switch from *weak caching* to *terse caching* with the s-maxage expiration value set 60 seconds by default. 
```
- mention terse caching
- there is no split view caching, so removed this

File: docs/deployment/caching/composite-views.md
Comment (jensens) [inline docs/deployment/caching/composite-views.md:19]: To increase the speed of a site that contains many composite views, a technique called "Terse" caching can be used. 
Terse caching involves caching a page for a brief period, typically 10 seconds in the browser and 60 seconds in the caching proxy. 
This ensures that the content is almost up to date, but during periods of high traffic, visitors may receive pages served from the cache.
This technique can actually speed up a site during periods of high traffic compared to low traffic since, during high traffic, the request can be served from the cache instead of having to go to the backend.

File: docs/deployment/caching/enable.md
Comment (jensens) [inline docs/deployment/caching/enable.md]:
```suggestion
To enable caching, click the checkbox:
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
ETags are used in to determine whether pages need to be re-calculated or can be served from cache.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
An ETag generated by `plone.app.caching' is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
This technique is used to quickly calculate the current ETag for the requested resource. 
The browser sends a request with the ETag included in an `If-None-Match` header, and if the ETag is the same, Plone responds with a 304 NOT MODIFIED message, indicating that the browser should use its cached copy. 
If the ETag is different, Plone renders the page and returns it as normal.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
Many caching operations use ETags. 
The tokens to include are typically listed in an `etags` tuple in the operation's options.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
The name is used to look up the component. 
Thus, you can also override one of the tokens above for a particular type of context or request (e.g. via a browser layer), by registering a more specific adapter with the same name.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md:58]: This generic adaption on Interface/Interface is a bad example.

File: docs/deployment/caching/index.md
Comment (jensens) [inline docs/deployment/caching/index.md]:
```suggestion
HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. 
When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.
```

File: docs/deployment/caching/index.md
Comment (jensens) [inline docs/deployment/caching/index.md]: Do we really want to mention Squid these days? 
Also here: split to one sentence per line
```suggestion
It is possible to use a web accelerator, like {term}`Varnish` to implement HTTP caching in your own premises. 
The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. 
If a cached copy is found, the accelerator will serve the cached copy directly to the client, otherwise it will make a request to the backend Plone server and then store a copy on the content in its local cache.
```

File: docs/deployment/caching/index.md
Comment (jensens) [inline docs/deployment/caching/index.md]:
```suggestion
HTTP caching and the speed of delivering web content can be improved using a Content Delivery Network (CDN).
A CDN is a network of servers located in various geographic regions that work together to deliver web content to users quickly and efficiently. 
When a user requests a resource from a website that uses a CDN, the request is directed to the closest server in the CDN, rather than having to travel all the way to the website's origin server.
```

File: docs/deployment/caching/index.md
Comment (jensens) [inline docs/deployment/caching/index.md]: I do not think we need a history lesson here ;)
```suggestion
Plone ships with a powerful and extensible `HTTP` and `In-Memory` cache support, implemented by the package `plone.app.caching`.
```

File: docs/deployment/caching/profiles.md
Comment (jensens) [inline docs/deployment/caching/profiles.md]:
```suggestion
`plone.app.caching` includes two default caching profiles.
These  profiles encapsulate the cache settings that are known to work well with a typical default Plone installation:
```

File: docs/deployment/caching/profiles.md
Comment (jensens) [inline docs/deployment/caching/profiles.md]:
```suggestion
Settings useful for setups with a caching proxy such Varnish or a CDN.
```

File: docs/deployment/caching/profiles.md
Comment (jensens) [inline docs/deployment/caching/profiles.md]: I am pretty sure this is not true anymore.

File: docs/deployment/caching/proxies.md
Comment (jensens) [inline docs/deployment/caching/proxies.md]:
```suggestion
This is the approach taken by Plone's resource management: Links that are inserted into Plone's content pages for resources managed by Plone  contain a hash-based token, which changes when the main bundle file changes.
```

File: docs/deployment/caching/proxies.md
Comment (jensens) [inline docs/deployment/caching/proxies.md]:
```suggestion
Then a PURGE request could be sent to the proxy (originating from the backend, not the browser) with the same path to force the proxy to fetch a new version the next time the item is requested.
```

File: docs/deployment/caching/proxies.md
Comment (jensens) [inline docs/deployment/caching/proxies.md]:
```suggestion
Synchronous and asynchronous purging is enabled via [plone.cachepurging](https://pypi.org/project/plone.cachepurging), which is installed as a dependency of `plone.app.caching`.
```

File: docs/glossary.md
Comment (jensens) [inline docs/glossary.md]: I would kick out Squid everywhere.

**Review by jensens — CHANGES_REQUESTED** [review]
Good to have the caching docs here now. Do not forget to remove them from plone.app.caching and set a link to the Plone docs. 

Overall there are many newlines between the sentences. This results in many paragraphs, is this by intend? 

Further, see my comments.

File: docs/deployment/caching/composite-views.md
Comment (jensens) [inline docs/deployment/caching/composite-views.md]:
```suggestion
To increase the speed of a site that contains many composite views, a technique called "Terse" caching can be used. 
Terse caching involves caching a page for a brief period, typically 10 seconds in the browser and 60 seconds in the caching proxy. 
This ensures that the content is almost up to date, but during periods of high traffic, visitors may receive pages served from the cache.
This technique can actually speed up a site during periods of high traffic compared to low traffic since, during high traffic, the request can be served from the cache instead of having to go to the backend.
```

File: docs/deployment/caching/etags.md
Comment (ericof) [reply docs/deployment/caching/etags.md:58]: Agreed. I just copied from the original documentation

File: docs/deployment/caching/profiles.md
Comment (ericof) [reply docs/deployment/caching/profiles.md]: I just checked the profiles, and it is it. Feed and Files use `moderateCaching` with `smaxage`:

## With Proxy
```xml
  <!-- plone.content.file -->
  <record name="plone.app.caching.moderateCaching.plone.content.file.smaxage">
    <field ref="plone.app.caching.moderateCaching.smaxage" />
    <value>86400</value>
  </record>
  <record name="plone.app.caching.moderateCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.moderateCaching.lastModified" />
    <value>True</value>
  </record>
```

## Without Proxy
```xml
  <record name="plone.app.caching.weakCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.weakCaching.lastModified" />
    <value>True</value>
  </record>
```

**Review by jensens — DISMISSED** [review]

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md]:
```suggestion
An ETag generated by `plone.app.caching` is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```

File: docs/deployment/caching/etags.md
Comment (jensens) [inline docs/deployment/caching/etags.md:15]: just found: "the current skin name" - is this nowadays a good example?

Comment (ericof) [reply docs/deployment/caching/etags.md:15]: This is still in `plone.app.caching`

**Comment by ericof** [comment]
@stevepiercy Any idea why the tests are failing?

**Comment by ksuess** [comment]
> @stevepiercy Any idea why the tests are failing?

Tests are green again after re-running.

**Comment by stevepiercy** [comment]
The PR from @fredvd that updated GitHub Actions broke the build after I merged it. I did not watch the build fail, and ignored the warning without checking its details, assuming it was ogp.me still failing under linkcheckbroken. I notified @fredvd at 5:10 AM Pacific, and he quickly responded by pushing [fixes to `6-dev`](https://github.com/plone/documentation/commits/6-dev). Everything should be good now. Sorry for the inconvenience.

Lesson learned: merge a PR only if you have time to watch it build successfully, and can fix it yourself if it fails.

**Comment by fredvd** [comment]
The weird part about why it failed was that yarn got upgraded to v3  during/after the PR.  so the $(yarn cache dir) no longer worked.  

Work on the workflows willstabose soonish, unfortunately small changes here can have big impact.

**Review by jensens — DISMISSED** [review]

**Comment by stevepiercy** [comment]
Please do not merge. I am doing a review, and will push fixes soon.

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (stevepiercy) [inline docs/deployment/caching/rulesets-and-caching-operations.md:63]: I don't see any of these default cache operations on the [Plone Classic Demo site](https://classic.demo.plone.org/@@caching-controlpanel#autotoc-item-autotoc-3). Where are they?

**Review by stevepiercy — COMMENTED** [review]
A question for now, while I continue my review.

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (stevepiercy) [inline docs/deployment/caching/rulesets-and-caching-operations.md]: Remove Archetypes and merge this sentence with the next one.

**Comment by stevepiercy** [comment]
My review is now complete. Please respond to my question and my suggestion. Thank you!

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: The demo site has caching installed, but not set up. I now set it up by importing the `Without caching proxy` profile and then enabled caching - then all is there.
(this will be auto-reset soon).

Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: OK, I see them now. This process needs to be documented. Newbies like me have no clue.

Here is a very rough outline, but to be honest, I have no idea what I'm doing. Maybe someone with more familiarity can fill in the details?

First, enable caching, from `caching/enable.md`.

Then:

https://classic.demo.plone.org/@@caching-controlpanel-import

Advanced > Import caching profiles > select Without caching proxy

Then:

https://classic.demo.plone.org/@@caching-controlpanel#autotoc-item-autotoc-4

Advanced > Caching settings > Detailed settings

Then:

- [View/edit](https://classic.demo.plone.org/@@caching-controlpanel/edit-operation-global/plone.app.caching.weakCaching) global parameters
- [View/edit/clear](https://classic.demo.plone.org/@@caching-controlpanel/edit-operation-ruleset/plone.app.caching.weakCaching/plone.content.feed) per-ruleset parameters

Comment (ericof) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: @stevepiercy There is a message on top of the page: 
`First time here? We recommend that you get started by importing a preconfigured set of caching rules`

Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: OK, but how does a newbie do that? This single sentence does not tell a newbie how to do it, which is why I suggested the above outline.

**Review by stevepiercy — DISMISSED** [review]
Please remove:

- `docs/_static/caching/ControlPanel-03.png`
- submodules

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: Detailed instructions are missing in `enable.md`, which we could refer to here.

Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: I'm fine with wherever it belongs. I need help fleshing out the details of the suggested outline.

Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: After (or better before) enable, use the controlpanle to import a profile:
screenshot_2023-02-25_23:24:02_selection

Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: forgot to say: no need to do more, no advanced/ZMI, nothing else, two steps. I usually import one of the profiles, depending on the setup and then click enable.

Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: Resolved in [87dcb50](https://github.com/plone/documentation/pull/1449/commits/87dcb5062e646f50d3486e371d0398bff6529133). Please review. Preview: https://63fb44f1bf95c20008d66c9d--6-docs-plone-org.netlify.app/deployment/caching/rulesets-and-caching-operations.html, then click through to new and existing sections.

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md]: This is the last blocker. I'm not sure of wording, but here is what I would suggest.

```
    In fact, `plone.app.caching` already does this for you.
    Dexterity `IDexterityItem` and `IDexterityContainer` interfaces are assigned the rulesets `plone.content.itemView` and `plone.content.folderview`, respectively.
```

@ericof @jensens Please let me know if this is in fact correct, and we can commit and merge this PR. Thank you!

File: docs/deployment/caching/enable.md
Comment (jensens) [inline docs/deployment/caching/enable.md]:
```suggestion
:   Settings useful for setups with a caching proxy such as Varnish or a CDN
```

File: docs/deployment/caching/enable.md
Comment (jensens) [inline docs/deployment/caching/enable.md]:
```suggestion
:   Settings useful for setups without a caching proxy such as Varnish or a CDN
```

File: docs/deployment/caching/profiles.md
Comment (jensens) [inline docs/deployment/caching/profiles.md]:
```suggestion
    description="Settings useful for setups with a caching proxy such as Varnish or a CDN"
```

**Review by jensens — CHANGES_REQUESTED** [review]
Do not mention outdated Squid.

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md:63]: good, except (see review below) the mention of Squid sneaked in again.

**Comment by stevepiercy** [comment]
@jensens @ericof please see the final blocker to merging this PR:

https://github.com/plone/documentation/pull/1449#discussion_r1118070026

File: docs/deployment/caching/rulesets-and-caching-operations.md
Comment (jensens) [reply docs/deployment/caching/rulesets-and-caching-operations.md]: Yes, see https://github.com/plone/plone.app.caching/blob/master/plone/app/caching/caching.zcml#L106-L114

Comment (stevepiercy) [reply docs/deployment/caching/rulesets-and-caching-operations.md]: Done in 0dab350b6ad27f3f4a307af95f4b46774e18eb06

**Comment by stevepiercy** [comment]
One more final review @jensens, if you please. Thank you!

**Review by jensens — APPROVED** [review]

**Pull request merged**
